### DISCUSSION TITLE
BUILD FAILS reports issue in Microsoft.Windows.UI.Xaml.Common.targets

### DISCUSSION THREAD

**Discussion by ToddGlodek** [unoplatform/uno#8145] [Q&A]
After compiling my UnoPlatform project successfully thousands of times, I have started getting a new compile/build error after splitting some of the project file out into a "Multi-Platform Library".   The error is not related to my own code but instead is some kind of "Xaml Internal Error WMC9999"  related to the file "Microsoft.Windows.UI.Xaml.Common.targets".  Specifically, the error is:

> Index (zero based) must be greater than or equal to zero and less than the size of the argument list.

I'm not an expert in this level of build mechanics detail, but this 'common targets file' appears is a file used by all build operations.   All my other (non-Uno) projects continue to work - and some of the projects in my Uno solution build successfully - just not this one library project.  

As you can see from the attached image, it is the second '.controls' library project that wont compile.   The '.controls' library project tries to reference another '.core' library  as a project reference - however, it displays a little 'ignored' icon in the Dependencies tree.  I read in your [online documentation on Working with cross-targeted class libraries](https://platform.uno/docs/articles/cross-targeted-libraries.html) that:

> Adding project references and framework references is not currently working through Visual Studio's interface is not currently working, it gives an error of "Missing value for TargetPlatformWinMDLocation property". You need to add the reference by editing the csproj file directly.

Is this the same issue ?  I have followed the instructions (even though it was a different error ).   Is what I am trying to do supported by Uno at this time.  The desire is to split the project up  so that different teams can work on different chunks of the application & can test their component independently before they are merged into the application as a whole.

![CommonTargetsBuildError](https://user-images.githubusercontent.com/19764509/154848759-3ee69121-6976-4e71-9e33-df8e4dd638b0.png)

**Comment by jeromelaban** [comment]
Thanks. Can you provide a binlog by [following these instructions](https://platform.uno/docs/articles/uno-builds-troubleshooting.html#generating-msbuild-binary-log-files) ?

**Comment by limeniye** [comment]
I think you can't add Core project for Controls project. But you can do it with .Net Standard 2.0. 
It happens because UAP doesn't support Core. 
You can use this version
```XML
  <ItemGroup>
    <ProjectReference Include="..\StandardTest\StandardTest.csproj" />
  </ItemGroup>
```
![image](https://user-images.githubusercontent.com/69314237/154851744-cace02b6-7338-453d-8ac2-4d7e19114589.png)

But if you need to use only Core project you can try next entry:
```XML
  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0-android' ">
    <ProjectReference Include="..\CoreTest\CoreTest.csproj" />
  </ItemGroup>
```

```C#
#if NETCOREAPP
using CoreTest;
#endif

namespace UnoTest
{
    public class Class1
    {
#if NETCOREAPP
        ClassTest gasg;
#endif
    }
}
```

**Reply by ToddGlodek** [reply]
Do you mean .Net Core project ?      I just happened to name the project "core" -- as far as I know its not a .Net Core project.   I will generate the 'binlog'  file as requested presently.

**Reply by limeniye** [reply]
Oh. Sorry, I misunderstood you.

**Reply by ToddGlodek** [reply]
Its an excellent answer (that will probably help somebody else) - I just worded the question unclearly
